### PR TITLE
Support selection of Unit of Measure for Temperature

### PR DESCRIPTION
**Pull request by phil-davis**
API enhanced to allow unit parameter in each query endpoint.
Results are returned with all temperatures converted to the requested
unit of measure.
The web app UI allows the user to choose the unit of measure, then
displays the selected unit of measure everywhere when
displaying/graphing the temperature.
Supports 4 temperature scales:
C - Celsius
F - Farenheit
K - Kelvin
R - Rankine (based off Farenheit but starting from absolute zero)

### CODE REVIEW

File: SERVER_README.md [hunk 1]
```diff
@@ -61,9 +68,27 @@ Example response:
...
+- `unit` - C, F, K or R return the result in Celsius, Farenheit, Kelvin or Rankine (default: C)
+
+Example response:
 
-Response follows the same schema as for the current temperature endpoint.
+```json
+{
+    "data": {
+        "unit": "C",
+        "spot": {
```
Comment (swalladge) [inline SERVER_README.md]: I would rather the scheme be:

```json
"data": {
  "unit": "C",
  "temperature": 00.0,
  "timestamp": 000000
}
```

File: database.py [hunk 2]
```diff
@@ -14,6 +14,19 @@
 
 Base = declarative_base()
 
+class TemperatureValue(object):
```
Comment (swalladge) [inline database.py]: Again, don't need any new classes!!! (This is python not java :P )

File: database.py [hunk 3]
```diff
@@ -25,9 +38,9 @@ def __repr__(self):
         return "Temperature {}°C at {}".\
                format(self.temperature, self.timestamp)
 
-    def dict(self):
+    def dict(self, unit):
         return {
-                'temperature': self.temperature,
+                'temperature': TemperatureValue(self.temperature).convert_temperature(unit),
                 'timestamp': self.timestamp
                 }
```
Comment (swalladge) [inline database.py]: can include a function here as a method of Temperature to convert the temperature value

**Comment by phil-davis** [comment]
Yes, fine. Take the stuff I have done here and massage it into the way you are thinking. I will be interested to see how you get all the conversions going inside the existing Temperature class.
I am learning Python and OOP!

**Comment by swalladge** [comment]
I have pushed an implementation to branch `units-test` - have a look and see what you think. If you like it, i'll merge it to master and you can rebase your webapp changes on it.

**Comment by phil-davis** [comment]
Server side implemented by #22 
Closing this and will submit another pull request for the web app.

**Pull request closed without merging**